src/core/helpers/notifications.ts

```diff
@@ -46,7 +47,7 @@ export const handleNotificationResponse = async (response) => {
 
   switch (type) {
     case BLENDED_COURSE_REGISTRATION:
-      return navigationRef.current?.navigate('BlendedAbout', { course });
+      return navigationRef.current?.navigate('BlendedAbout', { course: course as BlendedCourseType });
```

Comment:
J'ai l'impression que c'est pas recommander d'utiliser l'assertion : 
```
Type assertions are a way to tell the compiler “trust me, I know what I’m doing.” A type assertion is like a type cast in other languages, but it performs no special checking or restructuring of data. It has no runtime impact and is used purely by the compiler. TypeScript assumes that you, the programmer, have performed any special checks that you need.
https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions
```
Ya aussi cet article, qui a l'air contre les assertions,  cela dit l'article n'est pas claire car il parle d'un cas qui serait valide mais on comprend pas bien pq. 
https://medium.com/swlh/why-you-should-avoid-type-assertions-in-typescript-5494e3d04dd
----

J'ai l'impression que l'assertion équivaut  a un `// eslint-disable`. Alors qu'on pourrait utiliser un Type guard, https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards.

en gros 
```
// qq part dans le code (CoursTypes.tsx par ex)
function isBlended(course: CourseType): course is BlendedCourseType {
  return (course as BlendedCourseType).subProgram.isStrictlyELearning === false;
}


// helpers/notification.ts
return isBlended(course) ? navigationRef.current?.navigate('BlendedAbout', { course }) : null;
```

Reply:
non je vois rien qui donne plus d'info que la doc (https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards)

Rien sur un retour d'xp et bonnes pratiques

Reply:
le lient que tu as donné est deprecié : il vaut mieux lire celui la https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates :)